cime_config/config_component.xml

```diff
@@ -15,7 +15,7 @@
   <!-- Descriptions of all the different valid configurations for different model versions -->
   <description modifier_mode="1">
     <desc lnd="CLM45[%SP][%SP-VIC][%CN][%CNDV][%CN-CROP][%CNDV-CROP][%BGC][%BGC-CROP][%FATES][%BGCDV][%BGCDV-CROP]"                  >clm4.5:</desc>
-    <desc lnd="CLM50[%SP][%SP-VIC][%CN][%BGC][%BGC-CROP][%FATES][%BGCDV][%BGCDV-CROP][%BGC-CROP-CMIP6DECK][%BGC-CROP-CMIP6WACCMDECK]">clm5.0:</desc>
+    <desc lnd="CLM50[%SP][%SP-VIC][%CN][%BGC][%BGC-CROP][%FATES][%BGCDV][%BGCDV-CROP][%BGC-CROP-CMIP6DECK][%BGC-CROP-CMIP6WACCMDECK][%NWP-SP]">clm5.0:</desc>
```

Comment:
We need to make a choice here between combinatoric explosion (and the corresponding maintenance issues) vs. less error-checking at create_newcase time. The implementation here is the combinatoric explosion option, but I mainly chose that for now because it was slightly easier up-front, not because I'm necessarily in favor of that option.

With the current, combinatoric explosion implementation, we'll need to explicitly add support here for `%NWP-SP-VIC`, `%NWP-BGC`, `%NWP-BGC-CROP`, and many other options, assuming we want to support them. This will require an addition in the above line as well as a new line below for the description. We would have to do the same thing again to support a `%HYD` compset option that can be combined with these various other options.

The alternative would be to allow two (or more) `%` options on the compset name. So we'd have a compset like `CLM50%NWP%SP`. This would avoid the combinatoric problem. However, currently, I think the scripts would allow you to set up a case with `CLM50%NWP` without any BGC modifier (SP, BGC, etc.)... at least, I don't think this would abort at `create_newcase` time, but it might still be caught at `preview_namelists` time.

I guess my preference would be to see if we can have our cake and eat it, too: avoid the combinatoric explosion but see if we can come up with a fairly easy way of catching errors like `CLM50%NWP` without a BGC modifier.

What do others (especially @ekluzek ) think?

Reply:
Well, I think for now with just adding a single new option -- this is fine. 

I do like the concept of listing the valid configurations so it won't let you do an invalid option. I can see how we could trap this at `preview_namelist` time. But, I think you really want to catch it at `create_newcase` time. I think right now if we allowed more options, `CLM50%NWP` would use a default BGC option. So it would do something, but not necessarily what the user really wants. But, we could change it so that if you don't set it -- it aborts. To trap this at `create_newcase` time I think we need to add some more functionality into cime.

Reply:
I'm going to resolve this since we are going to go with the current implementation for now.